docs/internal/vmmap.md

```diff
@@ -0,0 +1,96 @@
+# Memory Management and Vmmap
+
+## What is a Vmmap?
+
+A vmmap is a tool for managing a process’s memory layout within an operating system. It provides detailed insights into allocated memory regions, including the heap, stack, and memory-mapped files. Additionally, it displays access permissions (read-only, read-write, executable), memory region boundaries, sizes, and any mapped files.
+
+## Motivation
+
+Wasmtime traditionally manages memory using WebAssembly’s linear memory model, where each instance gets a contiguous memory block divided into 64 KiB pages. This memory can grow or shrink dynamically within defined constraints. Since Lind emulates processes as cages within a single address space, tracking allocated memory regions per cage is essential.
+
+Attempts to provide POSIX-like interfaces for WASM, such as wasi-libc and emscripten, rely on WASM's memory.grow feature to expand available memory. Both implement custom malloc() functions that use memory.grow to extend the heap while preventing system mmap operations. Alternatively, they simulate file-backed mmap by invoking memory.grow and manually copying file contents into the allocated region.
+
+To address this, we eschew memory.grow and integrat a vmmap system into Lind that more closely resembles POSIX-based memory management. This allows proper implementation of syscalls like brk(), mmap(), and mprotect() for memory allocation, deallocation, and permission management. It also ensures accurate memory region copying when forking cages.
```

Comment:
A little more about "why" would be useful here.   Why is it useful to mimic POSIX-based memory management?   Why is this needed to ensure accurate memory region copying when forking?   

Should be about 1-2 more sentences for points like these.